lib/paperclip/storage/s3.rb

```diff
@@ -387,6 +387,10 @@ def flush_writes #:nodoc:
             write_options[:metadata] = @s3_metadata unless @s3_metadata.empty?
             write_options.merge!(@s3_headers)
 
+            sha256_checksum = Digest::SHA256.hexdigest(File.read(file.path))
+            encoded_sha256_checksum = Base64.encode64([sha256_checksum].pack("H*")).strip
+            write_options[:checksum_sha256] = encoded_sha256_checksum # Required for FIPS (SHA256) and object locking (checksum)
```

Comment:
@charchitarora isn't this already added by the aws-sdk-s3, is that what you meant by this is already being done

Reply:
@abdulchaudhrycoupa - I was referring to - https://github.com/coupa-ops/paperclip/blob/CE-14395_v5.2.0/lib/paperclip/attachment.rb#L36